lib/stimulus_reflex/logger.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+module StimulusReflex
+  class Logger
+    attr_accessor :reflex, :current_operation
+
+    COLORS = {
+      red: "31",
+      green: "32",
+      yellow: "33",
+      blue: "34",
+      magenta: "35",
+      cyan: "36",
+      white: "37"
+    }
+
+    def initialize(reflex)
+      @reflex = reflex
+      @current_operation = 1
+    end
+
+    def print
+      return unless config_logging.lambda?
+
+      puts
+      reflex.broadcaster.operations.each do
+        puts log + "\e[0m"
+        @current_operation += 1
+      end
+      puts
+    end
+
+    private
+
+    def config_logging
+      StimulusReflex.config.logging
+    end
+
+    def log
+      config_logging.call(self)
+    end
+
+    def session_id_full
+      session = reflex.request&.session
+      session.nil? ? "-" : session.id
+    end
+
+    def session_id
+      session_id_full.to_s[0..7]
+    end
+
+    def reflex_info
+      reflex.class.to_s + "#" + reflex.method_name
+    end
+
+    def reflex_id_full
+      reflex.reflex_id
+    end
+
+    def reflex_id
+      reflex_id_full[0..7]
+    end
+
+    def mode
+      reflex.broadcaster.to_s
+    end
+
+    def selector
+      reflex.broadcaster.operations[current_operation - 1][0]
+    end
+
+    def operation
+      reflex.broadcaster.operations[current_operation - 1][1]
+    end
+
+    def operation_counter
+      current_operation.to_s + "/" + reflex.broadcaster.operations.size.to_s
+    end
+
+    def connection_id_full
+      identifier = reflex.connection&.connection_identifier
+      identifier.empty? ? "-" : identifier
+    end
+
+    def connection_id
+      connection_id_full[0..7]
+    end
+
+    def timestamp
+      Time.now.strftime("%Y-%m-%d %H:%M:%S")
+    end
+
+    def method_missing method
+      return "\e[#{COLORS[method]}m" if COLORS.key?(method)
```

Comment:
I wouldn't use method missing to create the color methods, if we know which ones we create, I would use `def_method`

```ruby
COLORS.each do |name, code|
  define_method(name) { "\e[#{code}m" }
end
```

Reply:
Metaprogramming! I like this suggestion, Roland 👍